tree/treeplayer/inc/ROOT/TDFNodes.hxx

```diff
@@ -863,27 +865,27 @@ T &TColumnValue<T, B>::Get(Long64_t entry)
       // Currently we need the first entry to have been loaded to perform the check
       // TODO Move check to `MakeProxy` once Axel implements this kind of check in TTreeReaderArray using
       // TBranchProxy
-      if (!fArrayHasBeenChecked) {
-         if (readerArray.GetSize() > 1) {
-            if (1 != (&readerArray[1] - &readerArray[0])) {
-               std::string exceptionText = "Branch ";
-               exceptionText += readerArray.GetBranchName();
-               exceptionText += " hangs from a non-split branch. For this reason, it cannot be accessed via a TVec."
-                                " Please read the top level branch instead.";
-               throw std::runtime_error(exceptionText);
-            }
-            fArrayHasBeenChecked = true;
-         }
+
+      if (EStorageType::kUnknown == fStorageType && readerArray.GetSize() > 1) {
+         // We can decide since the array is long enough
+         fStorageType = (1 == (&readerArray[1] - &readerArray[0])) ? EStorageType::kContiguous : EStorageType::kSparse;
```

Comment:
i think we want the equivalent of `std::distance(&readerArray[1], &readerArray[0]) == 1` -- the distance between contiguous ptrs should be `sizeof(T)` or something like that

Reply:
argh, I don't know pointer arithmetics. the result is already in multiples of `sizeof(T)`, so this comment is totally wrong.